### PR TITLE
QA: show RPC integration test results on the run summary page

### PR DESCRIPTION
**Pull request by awskii**
## Summary
The QA RPC integration workflows dumped the raw `output.log` into the run's step summary, and the summary step hard-failed (`cat output.log`) when a run died during setup before producing a log — so the run page was blank exactly when it mattered (e.g. run 28475730212). And because the summary step only ran once tests had executed, a run that died in setup produced no summary at all — on the exec-mode matrix the run page then showed only the passing job and hid the failing one (run 28570398996: serial failed to sync, parallel passed, page showed only success).

This renders a clean summary onto the run page instead: the pass/fail result, an overall stats table, a table of the tests that failed, and the per-attempt retry history — without opening logs or downloading the artifact.

## Changes
- Add `.github/workflows/scripts/render_rpc_summary.py` (stdlib-only): reads the structured `results/test_report.json` and prints a result badge, a stats table (available / executed / passed / failed / not-executed / …), and a failed-tests table (test, transport, error). With no report it falls back to the badge plus a size-capped `output.log` tail. The pass/fail verdict comes only from the exit code and the structured report — never from parsing console text — so a runner output-format change cannot make it show a wrong result. The reporter never fails the job.
- Render a summary even when the run dies during setup (build / migrations / sync): the step now runs `if: always()` and falls back to `job.status` for `--result`, so a setup failure shows a `❌ failure` summary instead of nothing.
- Add an `Attempts` section scanned from the full `output.log` (the tests retry up to 3× and only the final attempt reaches the JSON; the log tail drops the earlier ones). It lists each attempt as `sync failed` / `N failing tests` / `passed`, so you can tell it passed on the first try vs only on retry. Informational only — never the verdict.
- Wire it into every QA RPC integration workflow: mainnet, gnosis, latest, remote, clients — and add a summary step to polygon, which had none.
- Add `render_rpc_summary.test.py`: fixture tests for the report, no-report, malformed-report, empty-log, setup-failure and multi-attempt paths. Run: `python3 .github/workflows/scripts/render_rpc_summary.test.py`.

## Notes
- The structured report is reliably produced once rpc-tests is repinned (#22133); until then the fallback shows the badge + raw log.
- Performance workflows are out of scope — they emit a different vegeta latency artifact, not a pass/fail test list.

### CODE REVIEW

**Review by lystopad — APPROVED** [review]

File: .github/workflows/qa-rpc-integration-tests-latest.yml [hunk 1]
```diff
@@ -217,29 +217,17 @@ jobs:
...
+          # run summary page. The reporter never fails the job and always prints
+          # something, even when the run died before producing a report.
+          SUMMARY=$(python3 "${{ github.workspace }}/.github/workflows/scripts/render_rpc_summary.py" \
+            --result-dir "${{ env.TEST_RESULT_DIR }}" \
+            --workflow "${{ github.workflow }}" \
+            --chain "${{ env.CHAIN }}" \
+            --result "${{ steps.test_step.outputs.TEST_RESULT }}")
+          printf '%s\n' "$SUMMARY"
+          printf '%s\n' "$SUMMARY" >> "$GITHUB_STEP_SUMMARY"
+          mkdir -p "${{ env.TEST_RESULT_DIR }}" 2>/dev/null && printf '%s\n' "$SUMMARY" > "${{ env.TEST_RESULT_DIR }}/summary.md" || true
```
Comment (lupin012) [inline .github/workflows/qa-rpc-integration-tests-latest.yml]: Now in case test latest The summary Report this data (example)

Attempt 1
Latest batch 1/5 (50 tests)
sync on latest block number failed  nodes not synced, testingNode=25427041 referenceNode=25427045 
Attempt 2
Latest batch 1/5 (50 tests)
0169. http           ::debug_traceBlockByNumber/test_24.json                          failed: cannot decode http body as json Read: unexpected value type: 0, error found in #10 byte of ...|,"result":,"error":{|..., bigger context ...|770401","result":,"error":{"code":-32000,"message":"tracing failed:|...
0187. http           ::debug_traceBlockByNumber/test_42.json                          failed: diff mismatch
0189. http           ::debug_traceBlockByNumber/test_44.json                          failed: diff mismatch
0192. http           ::debug_traceBlockByNumber/test_47.json                          failed: diff mismatch
0540. http           ::eth_blobBaseFee/test_01.json                                   failed: diff mismatch
0676. http           ::eth_feeHistory/test_07.json                                    failed: diff mismatch
Latest batch 1/5 had failures, stopping
Attempt 3
Latest batch 1/5 (50 tests)
Latest batch 2/5 (50 tests)
Latest batch 3/5 (50 tests)
Latest batch 4/5 (50 tests)
Latest batch 5/5 (17 tests)

I think this information was lost with the new script.
The tests are repeated up to 3 times. Each attempt runs all tests in batches of 50 tests
Knowing if all tests passed on the first or last attempt would be useful.
Example in this case:
Attempt 1: Fail sync
Attempt 2: few tests failed
Attempt 3: All tests are passed

Comment (awskii) [reply .github/workflows/qa-rpc-integration-tests-latest.yml]: i wasnt able to see any info before this update, file was not in place, and this consumes json output

Comment (awskii) [reply .github/workflows/qa-rpc-integration-tests-latest.yml]: Good point — the JSON report only reflects the final attempt. Added an `Attempts` section, scanned from the full `output.log` (not the tail, which drops the earlier attempts), listing each attempt as `sync failed` / `N failing tests` / `passed`. So your example renders as Attempt 1 ❌ sync failed · Attempt 2 ❌ 6 failing tests · Attempt 3 ✅ passed. Informational only — the verdict still comes from the exit code + JSON. 9b20354

File: .github/workflows/scripts/render_rpc_summary.py [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+            for i, f in enumerate(failed[:MAX_FAILURES], 1):
+                name = one_line(f.get("test_name", ""), 120)
+                transport = one_line(f.get("transport_type", ""), 20)
+                err = one_line(f.get("error_message", ""), ERROR_MSG_MAXLEN) or "—"
+                out.append(f"| {i} | {name} | {transport} | {err} |")
+            out.append("")
+            if len(failed) > MAX_FAILURES:
+                out += [f"> …and {len(failed) - MAX_FAILURES} more — see the `test-results` artifact.", ""]
+        elif (args.result or "").lower() != "failure":
+            out += ["✅ All executed tests passed.", ""]
```
[the remark this answers is not in the document]
Comment (awskii) [reply .github/workflows/scripts/render_rpc_summary.py]: Hardened: the all-passed line now requires explicit `success` **and** `summary.failed_tests == 0`, so an unknown/cancelled outcome, or a report that still counts failures while listing none, no longer shows a false pass. 9b20354

**Review by lupin012 — APPROVED** [review]

File: .github/workflows/qa-rpc-integration-tests-latest.yml (hunk 1 above)
Comment (awskii) [reply .github/workflows/qa-rpc-integration-tests-latest.yml]: The Attempts table now also lists the failing test names per attempt (up to five, with a +N overflow marker) in b5db48f640 — earlier-attempt failures only exist in the log, so this keeps them visible the way your example shows. The per-attempt outcome (sync failed / N failing / passed) was already in the table since the previous update.
